src/node.rs

```diff
@@ -1270,6 +1321,84 @@ mod tests {
         assert_eq!(state1.seq_num, 0);
     }
 
+    #[test]
+    fn block_new_first_10_blocks() {
```

Comment:
@aludvik, @ltseeley: In the interest of getting this merged I'm not going to add more unit tests beyond this to this PR, but it would be really useful to start writing some unit tests around various conditions such as catching up after having crashed without saving state.